lib/widgets/text.dart

```diff
@@ -286,6 +327,23 @@ FontWeight clampVariableFontWeight(double wght) {
   }
 }
 
+/// A "wght" value extracted from a [TextStyle].
+///
+/// Returns the value in [TextStyle.fontVariations] if present.
+/// If that's absent but a [TextStyle.fontWeight] is present,
+/// returns [wghtFromFontWeight] for that value.
+///
+/// The returned value already reflects any response to the system bold-text
+/// setting, so if the [TextStyle] was built using [weightVariableTextStyle],
+/// this value might be larger than the `wght` that was passed to that function.
+double? wghtFromTextStyle(TextStyle style) {
+  double? result = style.fontVariations?.singleWhereOrNull((v) => v.axis == 'wght')?.value;
```

Comment:
```suggestion
  double? result = style.fontVariations?.firstWhereOrNull((v) => v.axis == 'wght')?.value;
```

The "single" behavior means that if for some reason there are multiple `wght` variations, this would ignore them all and fall back to `fontWeight`. That doesn't seem intended.

I don't feel a strong need to have this code check that there aren't multiple of them. (In fact it looks like its one caller already has an assert to that effect.) If we're not going to check, then it should just do the simplest thing, which is to break the loop and return as soon as it finds one. That's what the "first" version does.

Reply:
Makes sense; I agree.